File: cliente_movil/src/types/visit.ts
export interface VisitMultimedia {
  id: string;
  visit_id: string;
  file_name: string;
  file_type: string;
  file_size: number;
  created_at: string;
  updated_at: string;
}

export interface Visit {
  id: string;
  nombre_institucion: string;
  direccion: string;
  hora: string;
  desplazamiento_minutos: number | null;
  hora_salida: string | null;
  estado: string;
  observacion: string | null;
  multimedia?: VisitMultimedia[];
  created_at: string;
  updated_at: string;
}

export interface MultimediaFile {
  uri: string;
  name: string;
  type: string;
}

export interface VisitCreate {
  nombre_institucion: string;
  direccion: string;
  hora: string;
  desplazamiento_minutos?: number;
  hora_salida?: string;
  estado: string;
  observacion?: string;
  files?: MultimediaFile[];
}

export interface VisitsResponse {
  data: Visit[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

File: backend/SalesForce/app/modules/visits/schemas/visit.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict

from .visit_multimedia import VisitMultimedia


class VisitBase(BaseModel):
    nombre_institucion: str
    direccion: str
    hora: datetime
    desplazamiento_minutos: Optional[int] = None
    hora_salida: Optional[datetime] = None
    estado: str
    observacion: Optional[str] = None


class VisitCreate(VisitBase):
    pass


class Visit(VisitBase):
    id: str
    created_at: datetime
    updated_at: datetime
    multimedia: Optional[List[VisitMultimedia]] = []

    model_config = ConfigDict(from_attributes=True)


class VisitsResponse(BaseModel):
    data: List[Visit]
    total: int
    page: int
    limit: int
    total_pages: int
